### ISSUE TITLE
x/pkgsite: code block scroll-in-scroll

### ISSUE THREAD

**Issue by BourgeoisBear** [golang/go#45073]
### What is the URL of the page with the issue?
 
https://pkg.go.dev/robpike.io/ivy
  
### What is your user agent?

Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36
 
### Screenshot

![image](https://user-images.githubusercontent.com/8608535/111404803-c7846300-86a5-11eb-92c0-508b9eae44aa.png)

### What did you do?

Trying to read the syntax documentation of Rob Pike's ivy.  The scroll-in-scroll of tall code blocks is painful.  A single set of scrollbars would be preferable:

![image](https://user-images.githubusercontent.com/8608535/111405037-2cd85400-86a6-11eb-816e-69df369b5006.png)

**Comment by Joanne881107** [comment]
Totally agreed that scroll-in-scroll of tall code blocks is painful. A few options here:
1. Since we do not have other horizon scrolling needs other than the code snippet, we can use the browser horizon scrolling for the code snippet as a quick fix.
2. As an UI enhancement, we can also consider enabling pop up the code in a full screen layer as an alternative option. Example here:
![Example](https://user-images.githubusercontent.com/8199471/116615497-3a613a80-a8f0-11eb-92be-d1528a7ad8e2.png)

**Comment by BourgeoisBear** [comment]
@Joanne881107 personal preference would be #1 as a permanent solution.  Simple, standard, and no accidental pop-ups from my touchpad.

**Comment by fflewddur** [comment]
I think we can and should avoid the need for visitors to do anything (either horizontal scrolling or viewing a pop up / overlay) in order to view this documentation. When the browser width is < 975px, we currently condense the details and outline, allowing the documentation to fill the full browser width. Above 975px, we immediately enable both the left and right sidebars. I suggest we add a third state, where the details sidebar continues to use the condensed horizontal view, and only switch to the 3-column layout when the center column will still be wide enough to display ~90 characters in preformatted boxes (which is what is causing the horizontal scrolling here).

The 90 character limit is arbitrary; it might be worth doing an analysis on the documentation corpus to identify a limit that matches current practices, but anything between 80-100 seems reasonable to me.

**Comment by fflewddur** [comment]
I think the above approach would also help with #44172.

**Comment by BourgeoisBear** [comment]
@fflewddur 

Not getting much value out of the 3rd column.  Most of it is empty.  If it's there just to aid in tracking the text, could just cap the width per-paragraph instead of the entire div.

Letting the code blocks be as long as they need to be (instead of scroll-in-scroll, or wrapping) would be a better experience IMHO.

![image](https://user-images.githubusercontent.com/8608535/116623797-cc7b3a00-a914-11eb-85a9-43ef05e9e798.png)

**Comment by fflewddur** [comment]
@BourgeoisBear I like the idea of letting the the code blocks run longer than they currently do, but we're trying to balance that with the width of non-code text, which does need a max width to support readability. So, even if there was no third column, we would still want to limit the maximum line length for regular text. If it's technically feasible to allow `<pre>` blocks to run longer than regular text, that's a design worth exploring.

**Comment by BourgeoisBear** [comment]
@fflewddur, I made a mock-up using the developer console:

![image](https://user-images.githubusercontent.com/8608535/116634583-7913e680-a92a-11eb-903c-22b36fd82c37.png)

Two columns instead of three, 2nd w/ width=auto, put max width on <p>, leave <pre> sized to content.  I set the max width for <p> a touch narrower.

The serif font is from my browser style overrides.  I think it reads easier than sans-serif, but that is an artifact, not a request.
